src/plugins/media/media.js

```diff
@@ -0,0 +1,166 @@
+        onEnded;
+    
+    document.addEventListener("impress:init", function(event) {
+        //root = event.target;
+        api = event.detail.api;
+        gc = api.lib.gc;
+        
+        enhanceMedia();
+        
+        gc.pushCallback(removeMediaClasses);
+    }, false);
+
+    onPlay = function (event) {
+        var type = event.target.nodeName.toLowerCase();
+        
+        document.body.classList.add('impress-media-' + type + "-playing");
+        document.body.classList.remove('impress-media-' + type + "-paused");
+    };
+
+    onPause = function (event) {
+        var type = event.target.nodeName.toLowerCase();
+        
+        document.body.classList.add('impress-media-' + type + "-paused");
+        document.body.classList.remove('impress-media-' + type + "-playing");
+    };
+
+    onEnded = function (event) {
+        var type = event.target.nodeName.toLowerCase();
+        
+        document.body.classList.remove('impress-media-' + type + "-playing");
+        document.body.classList.remove('impress-media-' + type + "-paused");
+    };
+    
+    removeMediaClasses = function () {
+        var types = ['video', 'audio'];
+        for (var type in types) {
+            document.body.classList.remove('impress-media-' + types[type] + "-playing");
+            document.body.classList.remove('impress-media-' + types[type] + "-paused");
+        }
+    }
+
+    enhanceMediaNodes = function () {
+        var i, id, media, type;
+        
+        //gc = event.detail.api.lib.gc;
+        media = document.querySelectorAll('audio, video');
+        for (i = 0; i < media.length; i++) {
+            type = media[i].nodeName.toLowerCase();
+            // Set an id to identify each media node - used e.g. by the consoleMedia plugin
+            id = media[i].getAttribute('id');
+            if (id === undefined || id === null) {
+                media[i].setAttribute('id', 'media-' + type + '-' + i);
+            }
+            gc.addEventListener(media[i], "play", onPlay);
+            gc.addEventListener(media[i], "playing", onPlay);
+            gc.addEventListener(media[i], "pause", onPause);
+            gc.addEventListener(media[i], "ended", onEnded);
+        }
+    };
+
+    enhanceMedia = function () {
+        var steps, i;
+        enhanceMediaNodes();
+        steps = document.getElementsByClassName('step');
+        for (i = 0; i < steps.length; i++) {
+            gc.addEventListener(steps[i], 'impress:stepenter', onStepenter);
+            gc.addEventListener(steps[i], 'impress:stepleave', onStepleave);
+        }
+    };
+
+    onStepenter = function (event) {
+        var media, play, i;
+        if ((!event) || (!event.target)) {
+            return;
+        }
+        removeMediaClasses();
+        media = event.target.querySelectorAll('audio, video');
+        for (i = 0; i < media.length; i++) {
+            play = media[i].dataset.mediaAutoplay;
+            if (play !== undefined && play !== "false") {
```

Comment:
Apologies. On second thought, this should be just:

    if ( play === "true" ) {

Anything else than the string "true" will then be false. This matches a similar attribute in impressConsole.js.

Reply:
Sure, but afaik it is common, to skip the true value of boolean values so that
`<video data-media-autoplay>` would be equivalent to `<video data-media-autoplay="true">` just as it is common to type `<video controls>` instead of `<video controls="true">`
Thus we would need 
```javascript
if (play === "" || play === "true")
```
This is compatible to impressConsole *and* html conventions.

Reply:
I learned HTML during the period where attributes were expected to always have a value (to make it well formed XML), but sure, I'm ok with that.